web_widget_numeric_step/static/src/numeric_step.esm.js

```diff
@@ -0,0 +1,89 @@
+/** @odoo-module */
+
+import {_t} from "@web/core/l10n/translation";
+import {registry} from "@web/core/registry";
+import {FloatField} from "@web/views/fields/float/float_field";
+import {standardFieldProps} from "@web/views/fields/standard_field_props";
+
+export class NumericStep extends FloatField {
+    setup() {
+        super.setup();
+    }
+    _onStepClick(ev) {
+        const mode = ev.target.dataset.mode;
+        this._doStep(mode);
+    }
+    _onKeyDown(ev) {
+        if (ev.keyCode === 38) {
+            this._doStep("plus");
+        } else if (ev.keyCode === 40) {
+            this._doStep("minus");
+        }
```

Comment:
When the user inputs a number directly using the keyboard, the minimum and maximum limits are not enforced. I think the behavior should be the same as when the user clicks the icons.

![image](https://github.com/user-attachments/assets/411ddb59-d47a-4d01-b04d-f1d9bc39032c)